Monster.m

```diff
@@ -15,68 +15,63 @@
 		Channel;
 		Traffic;
 		Results;
+		Logger;
 	end
 
 	methods 
-		function obj = Monster(Config)
+		function obj = Monster(Config, Logger)
 			% Monster constructor 
 			%
-			% :Config: MonsterConfig instance
-			% :Stations: Array<EvolvedNodeB> instances
-			% :Users: Array<UserEquipment> instances
-			% :Channel: SonohiChannel instance
-			% :Traffic: TrafficGenerator instance
-			% :Results: MetricRecorder instance
+			% :param Config: MonsterConfig instance
+			% :param Logger: MonsterLog instance
 
-			monsterLog('(MONSTER) setting up simulation', 'NFO');
-			obj.setupSimulation(Config);
-			monsterLog('(MONSTER) simulation setup completed', 'NFO');
+			obj.Logger = Logger;
+			obj.Config = Config;
+			obj.Logger.log('(MONSTER) setting up simulation', 'DBG');
+			obj.setupSimulation();
+			obj.Logger.log('(MONSTER) simulation setup completed', 'DBG');
 
 			if obj.Config.SimulationPlot.runtimePlot
 				% Draw the eNodeBs
-				obj.Config.Plot.Layout.draweNBs(obj.Config);
+				obj.Config.Plot.Layout.drawScenario(obj.Config);
 				% Draw the UEs
-				obj.Config.Plot.Layout.drawUes(obj.Users, obj.Config);
+				obj.Config.Plot.Layout.drawUes(obj.Users, obj.Config, obj.Logger);
 			end
 		end
 
-		function obj = setupSimulation(obj, Config)
+		function obj = setupSimulation(obj)
 			% setupSimulation calls the initialisation functions for the simulation properties
 			% 
 			% :param obj: Monster instance
 			% :returns obj: initialised Monster instance
 			%
-			
-			% Configure logs
-			setpref('monsterLog', 'logToFile', Config.Logs.logToFile);
-			setpref('monsterLog', 'logFile', Config.Logs.defaultLogName);
 
 			% Create network layout
-			monsterLog('(MONSTER - setupSimulation) setting up network layout', 'NFO');
-			Config.setupNetworkLayout();
+			obj.Logger.log('(MONSTER - setupSimulation) setting up network layout', 'DBG');
+			obj.Config.setupNetworkLayout(obj.Logger);
```

Comment:
It is indeed a bit of a challenge, to limit this, it has been changed so that it is only passed at the constructor level for relevant classes. This way a local reference to the main logger instance is used. In any case, good point, this has been changed to be more in line with the rest of the project style and because it was creating significant issues with parallel simulations. We should probably consider yet another solution in the near future.